### DISCUSSION TITLE
Tool section do not appear after open sirius session.

### DISCUSSION THREAD

**Discussion by Pitwng** [eclipse-sirius/sirius-desktop#485] [Q&A] [answered]
Hello ~
Dear Sirius Team:

Expected:
![image](https://github.com/user-attachments/assets/01a3c154-4657-4068-bf18-5cea7a897b25)

Using code open in RCP:
![image](https://github.com/user-attachments/assets/15411397-7ea1-4718-937b-e59e7bebd358)

Code:
```
		final URI resourceURI1 = URI.createFileURI("C:\\sample\\representations.aird");

		// Get the session
		final Session siriusSession = SessionManager.INSTANCE.getSession(resourceURI1, new NullProgressMonitor());

siriusSession.open(new NullProgressMonitor());
			final Resource sessionResource = siriusSession.getSessionResource();

			// Ensure the resource is not empty
			if (!sessionResource.getContents().isEmpty()) {
				final EObject root = sessionResource.getContents().get(0);
				if (root instanceof final DAnalysis analysis && !analysis.getOwnedViews().isEmpty()) {
					final DView dView = analysis.getOwnedViews().get(0);

					// Ensure the view contains representation descriptors
					if (!dView.getOwnedRepresentationDescriptors().isEmpty()) {
						final DRepresentationDescriptor myRepresentation = dView.getOwnedRepresentationDescriptors().get(0);

						// Open the representation in the editor
						final IEditorPart page = DialectUIManager.INSTANCE.openEditor(siriusSession, myRepresentation.getRepresentation(), new NullProgressMonitor());
						//						DialectUIManager.INSTANCE.refreshEditor((DialectEditor)page, null);
						// Ensure tools and layers are set correctly
						// Additional code to switch layers and set palette tools might be necessary here
					}
				}
			}
```

I add the page following this airticle: https://dirksmetric.wordpress.com/2012/08/01/tutorial-eclipse-rcp-e4-with-3-x-views-like-project-explorer-properties-etc/

BEST REGARDS.
Pitwng.

**Comment by GlennPlou** [comment]
Hello,
I'm not sure I understand what you're trying to do. Are you trying to open the editor programmatically? From what I understand, it's the Sirius palette that won't load. Considering the APIs you're using, I'd recommend you take a look at the junit or SWTBot tests we use, perhaps you'll find what you're missing there. At first glance, perhaps you could create and open the session with the following instructions:
```
session.open(new NullProgressMonitor());
SessionUIManager.INSTANCE.getOrCreateUISession(session).open();
```
Best regards,
Glenn

**Comment by Pitwng** [comment]
Hello
The Greatest Sirius Team in the Universe:
    I'm happy to tell you! I solve this problem. The reason the tool section didn't load correctly was that I overlooked the .odesign file. I failed to properly open the .odesign file, which caused the .aird file to be unable to access its resources. ChatGPT provided the solution. Obviously, this was a simple mistake, but fortunately, it's now resolved. I'll continue using the API you suggested to improve things (your advice probably aligns more with the standard). Thanks again for your response.
Best regards,
Pitwng

**Comment by lredor** [comment, marked as answer]
Alright, since the issue is resolved, I'll close this discussion.

**Comment by Pitwng** [comment]
Of course! Thank you !
